Jenkinsfile

```diff
@@ -1,15 +1,34 @@
 #!/usr/bin/groovy
 node{
   ws{
+
     stage 'checkout code'
-    checkout scm
+        checkout scm
+
+    stage 'build and unit test'
+    
+        kubernetes.pod('buildtestpod').withImage('maven')
+        .withPrivileged(true)
+        .withSecret('jenkins-maven-settings','/root/.m2')
+        .inside {
+            sh 'mvn -Dmaven.test.failure.ignore clean verify'
+            step([$class: 'JUnitResultArchiver', testResults: '**/target/surefire-reports/TEST-*.xml'])
+        }
 
     stage 'build and deploy'
-    kubernetes.pod('buildpod').withImage('maven')
-    .withPrivileged(true)
-    .withSecret('jenkins-maven-settings','/root/.m2')
-    .inside {
-        sh 'mvn clean deploy'
-    }
+        kubernetes.pod('deploypod').withImage('maven')
+        .withPrivileged(true)
+        .withSecret('jenkins-maven-settings','/root/.m2')
+        .inside {
+            try {
+                sh 'mvn clean deploy'
```

Comment:
* Deploy will deploy it to Nexus
* Clean I guess could be optional (force of habit)
* Required my Jenkins workflow to be able to capture tests results as per the documentation previously posted